developer-hub/src/reference_docs/extrinsic_encoding.rs

```diff
@@ -0,0 +1,156 @@
+//! # Constructing and Signing Extrinsics
+//!
+//! Substrate is configurable enough that extrinsics can take any format, in theory. In practice,
+//! runtimes tend to use our [`sp_runtime::generic::UncheckedExtrinsic`] type to represent extrinsics.
+//! In Polkadot, this is configured
+//! [here](https://github.com/polkadot-fellows/runtimes/blob/94b2798b69ba6779764e20a50f056e48db78ebef/relay/polkadot/src/lib.rs#L1478)
+//! at the time of writing.
+//!
+//! What follows is a description of how extrinsics based on this type are encoded into bytes. These
+//! bytes can then be submitted to a chain for inclusion in a block (this is how we make changes to
+//! the state of a chain), and are ultimately stored in the body of a block.
+//!
+//! # Encoding an Extrinsic
+//!
+//! At a high level, all extrinsics are formed from concatenating some details together, ie:
+//!
+//! ```text
+//! extrinsic_bytes = concat(
+//!     compact_encoded_length,
+//!     version_and_maybe_signature,
+//!     call_data
+//! )
+//! ```
+//!
+//! Let's look at how each of these details is constructed.
+//!
+//! ## compact_encoded_length
+//!
+//! This is a SCALE compact encoded integer which is equal to the length, in bytes, of the rest of
```

Comment:
Would be good to back this up by scale docs saying what is a company encoding etc.